File: cpp/code/EnvironmentController.cpp

#include "EnvironmentController.h"

void EnvironmentController::tic() {
    if (_hw->getTemp() < 16) {
        if (_hw->getTemp() < 6) {
            _hw->setLoTempAlarmOn();
        }
        _hw->turnBlowerOn();
        _hw->turnHeaterOn();
    }
    if (_hw->getTemp() > 24) {
        _hw->turnBlowerOn();
        _hw->turnCoolerOn();
        if (_hw->getTemp() > 29) {
            _hw->setHiTempAlarmOn();
        }
    }
}

File: java/src/main/java/codingdojo/EnvironmentController.java
package codingdojo;

public class EnvironmentController {
    private Hardware hw;

    public EnvironmentController(Hardware hw) {
        this.hw = hw;
    }

    public void tic() {
        if (hw.getTemp() < 16) {
            if (hw.getTemp() < 6) {
                hw.setLoTempAlarmOn();
            }
            hw.turnBlowerOn();
            hw.turnHeaterOn();
        }
        if (hw.getTemp() > 24) {
            hw.turnBlowerOn();
            hw.turnCoolerOn();
            if (hw.getTemp() > 29) {
                hw.setHiTempAlarmOn();
            }
        }
    }
}
